src/main/java/metrics_influxdb/JsonBuilderV9.java

```diff
@@ -0,0 +1,75 @@
+package metrics_influxdb;
+
+import java.util.Collection;
+import org.joda.time.DateTime;
+
+class JsonBuilderV9 implements JsonBuilder {
+	private final StringBuilder pointsJson = new StringBuilder();
+
+	private boolean hasSeriesData;
+
+	private String metadata;
+
+	public JsonBuilderV9(String database, String precision) {
+		metadata = "{\"database\":\"" + database + "\",\"precision\":\"" + precision + "\",";
+	}
+
+	@Override
+	public boolean hasSeriesData() {
+		return hasSeriesData;
+	}
+
+	@Override
+	public void reset() {
+		pointsJson.setLength(0);
+		pointsJson.append("\"points\":[");
+		hasSeriesData = false;
+	}
+
+	@Override
+	public String toJsonString() {
+		pointsJson.append(']');
+		String str = metadata + pointsJson.toString() + "}";
+		pointsJson.setLength(pointsJson.length() - 1);
+		return str;
+	}
+
+	@Override
+	public void appendSeries(String namePrefix, String name, String nameSuffix, String[] columns, Object[][] points) {
+		if (hasSeriesData)
+			pointsJson.append(',');
+		hasSeriesData = true;
+		final DateTime timestamp = new DateTime();
+		pointsJson.append("{\"name\":\"").append(namePrefix).append(name).append(nameSuffix).append("\",\"fields\":{");
+		Object[] row = points[0];
+		for (int j = 1; j < row.length; j++) {
+			if (j > 1)
+				pointsJson.append(',');
+			Object value = row[j];
+			pointsJson.append("\"" + columns[j] + "\":");
+			if (value instanceof String) {
+				pointsJson.append('"').append(value).append('"');
+			} else if ((value instanceof Collection) && ((Collection<?>) value).size() < 1) {
+				pointsJson.append("null");
+			} else if (value instanceof Double && Double.isInfinite((double) value)) {
+				pointsJson.append("null");
+			} else if (value instanceof Float && Float.isInfinite((float) value)) {
+				pointsJson.append("null");
+			} else {
+				pointsJson.append(value);
```

Comment:
You should either be much more restrictive about the types you accept here or you should sanitize the output of value's toString() function to avoid messing up the JSON with one bogus value.